app/code/core/Mage/SalesRule/Model/Validator.php

```diff
@@ -465,10 +465,10 @@ public function process(Mage_Sales_Model_Quote_Item_Abstract $item)
             $discountAmount = $result->getDiscountAmount();
             $baseDiscountAmount = $result->getBaseDiscountAmount();
 
-            $percentKey = $item->getDiscountPercent();
             /**
              * Process "delta" rounding
              */
+            $percentKey = (int)$item->getDiscountPercent();
```

Comment:
```suggestion
            $percentKey = (string) $item->getDiscountPercent();
```

Reply:
Isn't `$item->getDiscountPercent()` actually returns type string already? So, this won't do anything.

Reply:
getDiscountPercent() returns a float
<img width="504" alt="Screenshot 2023-07-14 alle 08 13 36" src="https://github.com/OpenMage/magento-lts/assets/909743/b7f0cdda-03d1-435e-b5b3-2383906a2782">

Reply:
I think the `@method` annotations are misleading here, it does indeed return a `string`:

![Screen Shot 2023-07-14 at 12 11 56 PM](https://github.com/OpenMage/magento-lts/assets/67818913/e8cb77ba-4796-4480-8123-d59bc400c777)

Reply:
mmmm could be, but then I think @kyrena opened this because they had the error of their systems

Reply:
and since, as a discount percent, it would be correct to have a float, but then it's used as an array key, I think we could force cast it to string just in case :-D

Reply:
Yes of course, `string` is the way to go.